lib/ruby_units/unit.rb

```diff
@@ -304,40 +318,23 @@ def self.base_units
     # parse a string consisting of a number and a unit string
     # NOTE: This does not properly handle units formatted like '12mg/6ml'
     # @param [String] string
-    # @return [Array] consisting of [Numeric, "unit"]
+    # @return [Array(Numeric, String)] consisting of [number, "unit"]
     def self.parse_into_numbers_and_units(string)
-      # scientific notation.... 123.234E22, -123.456e-10
-      sci       = /[+-]?\d*[.]?\d+(?:[Ee][+-]?)?\d*/
-      # rational numbers.... -1/3, 1/5, 20/100, -6 1/2, -6-1/2
-      rational  = %r{\(?[+-]?(?:\d+[ -])?\d+\/\d+\)?}
-      # complex numbers... -1.2+3i, +1.2-3.3i
-      complex   = /#{sci}{2,2}i/
-      anynumber = /(?:(#{complex}|#{rational}|#{sci}))?\s?([^-\d\.].*)?/
-
-      num, unit = string.scan(anynumber).first
+      num, unit = string.scan(ANY_NUMBER_REGEX).first
 
       [
         case num
-        when NilClass
+        when nil # This happens when no number is passed and we are parsing a pure unit string
           1
-        when complex
-          if num.respond_to?(:to_c)
-            num.to_c
-          else
-            #:nocov_19:
-            Complex(*num.scan(/(#{sci})(#{sci})i/).flatten.map(&:to_i))
-            #:nocov_19:
-          end
-        when rational
-          # if it has whitespace, it will be of the form '6 1/2'
-          if num =~ RATIONAL_NUMBER
-            sign = Regexp.last_match(1) == '-' ? -1 : 1
-            n = Regexp.last_match(2).to_i
-            f = Rational(Regexp.last_match(3).to_i, Regexp.last_match(4).to_i)
-            sign * (n + f)
-          else
-            Rational(*num.split('/').map(&:to_i))
-          end
+        when COMPLEX_NUMBER
```

Comment:
Use the same Regexp everywhere. Also we dropped the special handling for complex numbers because that hasn't been necessary for a very long time.